experimental/FTheoryTools/src/auxiliary.jl

```diff
@@ -199,21 +209,21 @@ function _kodaira_type(id::MPolyIdeal{T}, f::T, g::T, d::T, ords::Tuple{Int64, I
         kod_type = "Non-split I^*_0"
       end
     elseif f_ord == 2 && g_ord == 3 && d_ord >= 7 && d_ord % 2 == 1
-      monodromy_poly = _psi^2 + divexact(evaluate(divexact(poly_d, locus^d_ord) * divexact(2 * poly_f, locus^2)^3, [locus], [0]), 4 * evaluate(divexact(9 * poly_g, locus^3), [locus], [0])^3)
+      monodromy_poly = _psi^2 + divexact(evaluate(divexact(poly_d, locus^d_ord) * divexact(2 * weierstrass_f, locus^2)^3, [locus], [0]), 4 * evaluate(divexact(9 * weierstrass_g, locus^3), [locus], [0])^3)
       if _count_factors(monodromy_poly) == 2
         kod_type = "Split I^*_$(d_ord - 6)"
       else
         kod_type = "Non-split I^*_$(d_ord - 6)"
       end
     elseif f_ord == 2 && g_ord == 3 && d_ord >= 8 && d_ord % 2 == 0
-      monodromy_poly = _psi^2 + divexact(evaluate(divexact(poly_d, locus^d_ord) * divexact(2 * poly_f, locus^2)^2, [locus], [0]), evaluate(divexact(9 * poly_g, locus^3), [locus], [0])^2)
+      monodromy_poly = _psi^2 + divexact(evaluate(divexact(poly_d, locus^d_ord) * divexact(2 * weierstrass_f, locus^2)^2, [locus], [0]), evaluate(divexact(9 * weierstrass_g, locus^3), [locus], [0])^2)
       if _count_factors(monodromy_poly) == 2
         kod_type = "Split I^*_$(d_ord - 6)"
       else
         kod_type = "Non-split I^*_$(d_ord - 6)"
       end
```

Comment:
4 appearances of almost identical code. This multiplication should be eliminated by adding a small helper function. Maybe it can even swallow the other less similar blocks. If there is less code, it is easier to maintain.

Reply:
Just pushed a commit with this. Thank you for the suggestion @lkastner.